Go SDK - Part 2

aludvik opened the pull request:
Resolve TODO comments in code, add logging, and make a couple changes to the stream interface.

vaporos commented on sdk/go/src/sawtooth_sdk/processor/state.go:
```diff
@@ -8,18 +8,35 @@ import (
+// Get queries the validator state for data at each of the addresses in the
+// given slice. A string->[]byte map is returned. If an address is not set, the
+// slice in the map will have 0 length.
+//
```
Maybe add a "For example" here, so it looks less like commented-out code.

vaporos commented on sdk/examples/intkey_go/src/sawtooth_intkey/handler/handler.go:
```diff
@@ -4,12 +4,16 @@ import (
 	"crypto/sha512"
 	"encoding/hex"
 	"fmt"
+    "os"
```
Looks like spaces here, we probably need gofmt on these files.  We need to figure out how to add this to run_lint.

vaporos commented on sdk/examples/intkey_go/src/sawtooth_intkey/handler/handler.go:
```diff
@@ -4,12 +4,16 @@ import (
 	"strings"
 )
 
+var LOGGER *logging.Logger = logging.New(os.Stdout, logging.DEBUG)
```
Can we keep the API log4j-like where we set it up to have separation between logger and handlers?  This would then just take a string.

vaporos replied:
Should LOGGER be lower-case so it's not exported out of the module?

vaporos commented on sdk/go/src/sawtooth_sdk/processor/processor.go:
```diff
@@ -2,13 +2,19 @@ package processor
 
 import (
 	"fmt"
+    "os"
```
looks like spaces?

vaporos commented on sdk/go/src/sawtooth_sdk/processor/processor.go:
```diff
@@ -23,72 +29,76 @@ func NewTransactionProcessor(url string) *TransactionProcessor {
 				response.Status = processor_pb2.TpProcessResponse_INTERNAL_ERROR
 			default:
-				fmt.Println("Unknown error")
+				LOGGER.Warn("Unknown error")
```
If it implements Error, we could still print out the message.

peterschwarz replied:
This should also be logged as an error, since the developer will probably want to do something about it.

vaporos commented on sdk/go/src/sawtooth_sdk/processor/state.go:
```diff
@@ -8,18 +8,35 @@ import (
+// given slice. A string->[]byte map is returned. If an address is not set, the
+// slice in the map will have 0 length.
+//
+//     results, err := state.Get(addresses)
```
Perhaps add a "For example:"

peterschwarz commented on sdk/go/src/sawtooth_sdk/processor/handler.go:
```diff
@@ -1,13 +1,46 @@
 
+// TransactionHandler is the interface that defines the business logic for a
+// new transaction family. This is the only interface that needs to be
+// implemented to create a new transaction family!
```
Love the enthusiasm, but s/!/./

peterschwarz commented on sdk/go/src/sawtooth_sdk/processor/state.go:
```diff
@@ -72,6 +89,19 @@ func (self *State) Get(addresses []string) (map[string][]byte, error) {
+// Set requests that each address in the validator state be set to the given
+// value. A slice of addresses set is returned or an error if there was a
+// problem setting the addresses.
+//
```
Echoing @vaporos "For Example" comment for this block.

peterschwarz commented on sdk/go/src/sawtooth_sdk/messaging/stream.go:
```diff
@@ -136,14 +169,16 @@ func (self *Stream) start() {
 		return nil
 	}
 
+	// 2. Deserialize
```
Probably can just lose the numbers.  It has the natural ordering, and then it doesn't have to be updated if future steps are added in the middle.

aludvik commented:
1. Added "For example" as needed
2. Added Go linting to pipeline
3. Made logging more like log4j and changed cases
4. Removed numbering/enthusiasm from comments
5. Changed/added a couple logging statements
6. Added copyright headers

ojalatodd commented on sdk/go/src/sawtooth_sdk/processor/handler.go:
```diff
@@ -1,13 +1,63 @@
+	// Eg., "application/cbor"
 	Encoding() string
+
+	// Namespaces should return a slice containing all the handlers namespaces.
```
**Punctuation:** handlers --> handler's

ojalatodd commented on sdk/go/src/sawtooth_sdk/messaging/stream.go:
```diff
@@ -20,11 +43,23 @@ type sendRequest struct {
 	IsResponse bool
 }
 
+// Response is used to wrap a validator message coming of a channel returned by
```
Spelling?
" Response is used to wrap a validator message coming **of** a channel returned by" 

Did you mean "off" a channel, or "from a channel"?

ojalatodd commented on sdk/go/src/sawtooth_sdk/messaging/stream.go:
```diff
@@ -105,107 +164,132 @@ func (self *Stream) Respond(t validator_pb2.Message_MessageType, c []byte, corrI
+	return request.Response
 }
 
+// Setup the sockets and channels needed for communicating.
```
Usage: Setup is a noun, "set up" is a verb/imperative.
Rephrase: "Set up the sockets..."

ojalatodd commented on sdk/go/src/sawtooth_sdk/messaging/stream.go:
```diff
@@ -105,107 +164,132 @@ func (self *Stream) Respond(t validator_pb2.Message_MessageType, c []byte, corrI
+// Receive a single message and route
 func (self *Stream) receiver(socketState zmq.State) error {
-	// 1. Receive message
+	// Handle a panic while received without crashing. For example, if the
```
Usage? 
Change : Handle a panic while received without
to
 Handle a panic while receiving without

aludvik commented:
Addressed Todd's comments.
